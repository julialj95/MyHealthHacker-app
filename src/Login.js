import React from "react";

function Login() {
  return (
    <div className="login">
      <h1>LOG IN</h1>
      <h2 className="demo_access">
        *To test out MyHealthHacker with a demo account, log in with:
        <br />
        Username: demouser
        <br />
        Password: demopassword
      </h2>
      <form className="login_form">
        <input
          className="login_field"
          type="text"
          name="username"
          placeholder="Enter username"
        />

        <br />

        <input
          className="login_field"
          type="text"
          name="password"
          placeholder="Enter password"
        />

        <br />
        <button className="login_btn" type="submit">
          Log In
        </button>
      </form>
    </div>
  );
}
export default Login;
