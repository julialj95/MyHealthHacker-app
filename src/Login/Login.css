.login {
  text-align: center;
  margin: 10%;
  border: solid 3px #88d317;
  border-radius: 15px;
  background-color: #6e3667;
}

.login_form {
  padding-bottom: 15px;
}

.demo_access {
  color: white;
  padding-left: 10px;
  padding-right: 10px;
}

.login_header {
  color: white;
}

@media all and (min-width: 600px) {
  .login_header {
    font-size: 26px;
  }
  .demo_access {
    font-size: 22px;
  }
}
@media all and (min-width: 960px) {
  .login_header {
    font-size: 32px;
  }
  .demo_access {
    font-size: 28px;
  }
}
