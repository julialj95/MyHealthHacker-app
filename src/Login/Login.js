import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import TokenService from "../services/token-service";
import AuthService from "../services/auth-service";
import userContext from "../UserContext";
import "./Login.css";
function Login(props) {
  const context = useContext(userContext);
  const history = useHistory();
  const [userData, setUserData] = useState({ username: "", password: "" });
  const [error, setError] = useState("");

  const handleChange = (e) => {
    const { name, value } = e.target;
    setUserData({
      ...userData,
      [name]: value,
    });
  };

  const handleJwtAuthSubmit = (e) => {
    e.preventDefault();
    setError("");
    const { username, password } = userData;

    AuthService.postLogin({
      username: username,
      password: password,
    })
      .then((res) => {
        setUserData({ username: "", password: "" });
        TokenService.saveAuthToken(res.authToken);
        history.push("/submit-log");
        props.handleLogin(TokenService.hasAuthToken());
        context.setUsername(username);
      })
      .catch((error) => {
        setError(error.error);
      });
  };

  return (
    <main className="login">
      <h2 className="login_header">LOG IN</h2>
      <h3 className="demo_access">
        *To test out MyHealthHacker with a demo account, log in with the
        following credentials:
        <br />
        <br />
        Username: demouser
        <br />
        Password: demopassword
        <br />
      </h3>
      <form className="login_form" onSubmit={(e) => handleJwtAuthSubmit(e)}>
        <input
          className="login_field"
          type="text"
          name="username"
          placeholder="Enter username"
          onChange={(e) => handleChange(e)}
        />

        <br />

        <input
          className="login_field"
          type="text"
          name="password"
          placeholder="Enter password"
          onChange={(e) => handleChange(e)}
        />

        <br />
        <button className="form_btn" type="submit">
          Log In
        </button>
        {error ? (
          <h2>Incorrect username or password. Please try again.</h2>
        ) : null}
      </form>
    </main>
  );
}
export default Login;
