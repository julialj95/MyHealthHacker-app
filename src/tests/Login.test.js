import React from "react";
import ReactDOM from "react-dom";
import Login from "../Login/Login";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <Login />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <Login />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
