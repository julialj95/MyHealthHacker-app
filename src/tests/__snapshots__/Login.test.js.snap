// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="login"
>
  <h2
    className="login_header"
  >
    LOG IN
  </h2>
  <h3
    className="demo_access"
  >
    *To test out MyHealthHacker with a demo account, log in with the following credentials:
    <br />
    <br />
    Username: demouser
    <br />
    Password: demopassword
    <br />
  </h3>
  <form
    className="login_form"
    onSubmit={[Function]}
  >
    <input
      className="login_field"
      name="username"
      onChange={[Function]}
      placeholder="Enter username"
      type="text"
    />
    <br />
    <input
      className="login_field"
      name="password"
      onChange={[Function]}
      placeholder="Enter password"
      type="text"
    />
    <br />
    <button
      className="form_btn"
      type="submit"
    >
      Log In
    </button>
  </form>
</main>
`;
